oak-store-document/src/main/java/org/apache/jackrabbit/oak/plugins/document/DocumentNodeStore.java

```diff
@@ -2527,8 +2527,8 @@ private void backgroundSplit() {
             invalidatePaths(pathsToInvalidate);
             batchSplit(splitOpsPhase1);
             batchSplit(splitOpsPhase2);
-            splitCandidates.keySet().removeAll(removeCandidates);
         }
+        splitCandidates.keySet().removeAll(removeCandidates);
```

Comment:
This fixes indeed a regression, nice catch! (The above change fixes an older bug, that was there before)